multiversx_sdk_cli/cli_delegation.py

```diff
@@ -442,162 +632,528 @@ def _check_if_either_bls_keys_or_validators_file_are_provided(args: Any):
+        gas_limit=gas_limit,
+        gas_price=int(args.gas_price),
+        value=int(args.value),
+        nonce=nonce,
+        version=int(args.version),
+        options=int(args.options),
+        guardian_account=guardian,
+        guardian_address=guardian_address,
+        relayer_account=relayer,
+        relayer_address=relayer_address,
+        guardian_service_url=args.guardian_service_url,
+        guardian_2fa_code=args.guardian_2fa_code,
+    )
+
     cli_shared.send_or_simulate(tx, args)
 
 
 def unjail_nodes(args: Any):
     _check_if_either_bls_keys_or_validators_file_are_provided(args)
-    ensure_arguments_are_provided_and_prepared(args)
+    validate_transaction_args(args)
+    ensure_wallet_args_are_provided(args)
+    validate_broadcast_args(args)
+    validate_chain_id_args(args)
 
-    sender = cli_shared.prepare_account(args)
-    config = TransactionsFactoryConfig(args.chain)
+    sender, nonce = prepare_sender(args)
+    guardian, guardian_address = prepare_guardian(args)
+    relayer, relayer_address = prepare_relayer(args)
+
+    chain_id = cli_shared.get_chain_id(args.chain, args.proxy)
+    config = TransactionsFactoryConfig(chain_id)
     delegation = DelegationOperations(config)
 
-    tx = delegation.prepare_transaction_for_unjailing_nodes(sender, args)
+    gas_limit = 0 if args.estimate_gas else args.gas_limit
+    public_keys = _load_validators_public_keys(args)
+
+    tx = delegation.prepare_transaction_for_unjailing_nodes(
+        owner=sender,
+        delegation_contract=Address.new_from_bech32(args.delegation_contract),
+        public_keys=public_keys,
+        gas_limit=gas_limit,
+        gas_price=int(args.gas_price),
+        value=int(args.value),
+        nonce=nonce,
+        version=int(args.version),
+        options=int(args.options),
+        guardian_account=guardian,
+        guardian_address=guardian_address,
+        relayer_account=relayer,
+        relayer_address=relayer_address,
+        guardian_service_url=args.guardian_service_url,
+        guardian_2fa_code=args.guardian_2fa_code,
+    )
+
     cli_shared.send_or_simulate(tx, args)
 
 
 def delegate(args: Any):
-    ensure_arguments_are_provided_and_prepared(args)
+    validate_transaction_args(args)
+    ensure_wallet_args_are_provided(args)
+    validate_broadcast_args(args)
+    validate_chain_id_args(args)
 
-    if not (int(args.value)):
-        raise errors.BadUrlError("Value not provided. Minimum value to delegate is 1 EGLD")
+    sender, nonce = prepare_sender(args)
+    guardian, guardian_address = prepare_guardian(args)
+    relayer, relayer_address = prepare_relayer(args)
 
-    sender = cli_shared.prepare_account(args)
-    config = TransactionsFactoryConfig(args.chain)
+    gas_limit = 0 if args.estimate_gas else args.gas_limit
+
+    value = int(args.value)
+    if value < MINIMUM_AMOUNT_TO_DELEGATE:
+        errors.BadUserInput(f"Value must be greater than {MINIMUM_AMOUNT_TO_DELEGATE} (1 EGLD)")
```

Comment:
Check can be dropped :thought_balloon: (time-dependent).

Reply:
deleted